### PR TITLE
feat(core): internals controller improvements

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
- proxy `role` and `aria*` props in `ElementInternals` to `InternalsController`
- wrap the host's `form*Callback` methods in `InternalsController`
- add `formDisabled: boolean` to `InternalsController`
- disable BaseButton if it's disabled either by attr or by form, without requiring the reflected attr when form is disabled.

## Testing Instructions
Check out button's `form controls` demo and make sure everything's kosher there

### CODE REVIEW

File: core/pfe-core/controllers/internals-controller.ts [hunk 1]
```diff
@@ -1,12 +1,85 @@
...
+
+  static protos = new WeakMap();
+
   constructor(
     public host: ReactiveControllerHost & HTMLElement,
+    options?: Partial<ARIAMixin>
   ) {
     this.#internals = host.attachInternals();
+    // proxy the internals object's aria prototype
+    for (const key of Object.keys(this.#internals)) {
```
Comment (heyMP) [inline core/pfe-core/controllers/internals-controller.ts:64]: I can't get this `Object.keys(this.#internals)` to return anything but an empty array.  Is this working for you?

File: core/pfe-core/controllers/internals-controller.ts [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
+  declare ariaValueMax: ARIAMixin['ariaValueMax'];
+  declare ariaValueMin: ARIAMixin['ariaValueMin'];
+  declare ariaValueNow: ARIAMixin['ariaValueNow'];
+  declare ariaValueText: ARIAMixin['ariaValueText'];
+
   #internals: ElementInternals;
 
+  /** True when the control is disabled via it's containing fieldset element */
+  get formDisabled() {
+    return this.host.matches(':disabled');
```
Comment (heyMP) [inline core/pfe-core/controllers/internals-controller.ts:53]: Will this trigger a re-render for the host element?

File: core/pfe-core/controllers/internals-controller.ts [hunk 3]
```diff
@@ -1,12 +1,85 @@
 import type { ReactiveController, ReactiveControllerHost } from 'lit';
 
-export class InternalsController implements ReactiveController {
+function isARIAMixinProp(key: string): key is keyof ARIAMixin {
+  return key === 'role' || key.startsWith('aria');
+}
+
+export class InternalsController implements ReactiveController, ARIAMixin {
+  declare role: ARIAMixin['role'];
```
Comment (heyMP) [inline core/pfe-core/controllers/internals-controller.ts:8]: TS question.  Is there a reason we hare to `declare` these when we have already indicated that this class implements `ARIAMixin`

File: core/pfe-core/controllers/internals-controller.ts (hunk 2 above)
Comment (bennypowers) [reply core/pfe-core/controllers/internals-controller.ts:53]: No, see above comments. It's a platform limitation, you can't hijack the callback at runtime. Alternatives are a matching class decorator or a mixin.

File: core/pfe-core/controllers/internals-controller.ts (hunk 3 above)
Comment (bennypowers) [reply core/pfe-core/controllers/internals-controller.ts:8]: `implements` just tells the compiler to complain if they aren't there

File: core/pfe-core/controllers/internals-controller.ts (hunk 1 above)
Comment (bennypowers) [reply core/pfe-core/controllers/internals-controller.ts:64]: Maybe we should switch it back to reading the prototype

Comment (bennypowers) [reply core/pfe-core/controllers/internals-controller.ts]: this should be fixed now

File: core/pfe-core/controllers/internals-controller.ts (hunk 2 above)
Comment (bennypowers) [reply core/pfe-core/controllers/internals-controller.ts:53]: FYI this is what that would look like

```ts
import type { ReactiveElement } from 'lit';
import { InternalsController } from '../controllers/internals-controller.js';

function isReactiveElementClass(
  klass: Function // eslint-disable-line @typescript-eslint/ban-types
): klass is typeof ReactiveElement {
  return typeof klass['addInitializer' as keyof typeof klass] === 'function';
}

const instances = new WeakMap();

/**
 */
export function internals(instance?: ReactiveElement): InternalsController
export function internals(options?: Partial<ARIAMixin>): ClassDecorator
export function internals(optsOrInstance: unknown): InternalsController|ClassDecorator {
  if (optsOrInstance instanceof HTMLElement) {
    return instances.get(optsOrInstance);
  } else {
    return function(klass) {
      if (!isReactiveElementClass(klass)) {
        throw new Error(`@internals may only decorate ReactiveElements. ${klass.name} is does not implement ReactiveElement.`);
      }

      klass.addInitializer(instance => {
        const controller = new InternalsController(instance, optsOrInstance);
        instance.addController(controller);
        instances.set(instance, controller);
      });
    };
  }
}
```

```ts
@internals()
export class BaseButton extends LitElement {
  // ...
  #internals = internals(this);
  // ...
}
```

A mixin is arguably more honest here

Comment (heyMP) [reply core/pfe-core/controllers/internals-controller.ts:53]: I see what you're saying.  So end users would be required to use a mixin over a controller.

**Review by heyMP — APPROVED** [review]
👍

**Pull request merged**